### PR TITLE
Add mockStream() util

### PR DESCRIPTION
**Pull request by macsj200**
## 📚 Purpose
Add a utility to mock stream interfaces.

## 👌 Resolves:
- [ ] 🐛 AIR-110 fixes bug

## 📦 Impacts:
*[mds-stream]*

## ✅ PR Checklist
- [ ] Add or remove checklist items to suit your needs

Do I need to explicitly export the `WriteStream` type in `index.ts`?
Do I need to explicitly add `sinon` to deps?

### CODE REVIEW

**Comment by avatarneil** [comment]
Re: explicit export of WriteStream -- Yeah, I think that makes sense.

Re: `sinon` to dep -- I think you'll have to, while it seems kind of odd.

File: packages/mds-stream/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Sinon from 'sinon'
+import { WriteStream } from './types'
+
+export const mockStream = <T>(stream: WriteStream<T>) => {
```
Comment (avatarneil) [inline packages/mds-stream/test-utils.ts:4]: Would be cool if you could pass in an `options` or `overrides` object where you could supply methods for `initialize, shutdown, write`. E.g. `{ initialize: () => throw new Error(...) }` if you wanted to test for initialize failures, for example

File: packages/mds-stream/types.ts [hunk 2]
```diff
@@ -29,3 +29,9 @@ export type StreamItemType = string
 export type StreamItemData = string
 export type StreamItem = [StreamItemID, [StreamItemType, StreamItemData]]
 export type ReadStreamResult = [Stream, StreamItem[]]
+
+export interface WriteStream<T> {
```
Comment (avatarneil) [inline packages/mds-stream/types.ts:33]: Come to think of it -- I think you can just use the `StreamProducer` type!

Comment (macsj200) [reply packages/mds-stream/types.ts]: yeet

**Pull request merged**